### PR TITLE
use /mos-action file in rootfs to decide what to do with pcr7

### PR DESCRIPTION
**Pull request by hallyn**

### CODE REVIEW

File: layers/build-krd/dracut/soci/module-setup.sh [hunk 1]
```diff
@@ -31,6 +31,12 @@ install() {
     inst "/lib/udev/cdrom_id"
     inst "/lib/udev/rules.d/60-cdrom_id.rules"
     inst mknod
+    inst_multiple \
+        tpm2_createprimary tpm2_flushcontext tpm2_getcap tpm2_loadexternal \
+        tpm2_nvread tpm2_nvreadpublic tpm2_pcrextend tpm2_pcrread \
+        tpm2_policyauthorize tpm2_policynv tpm2_policypcr \
+        tpm2_startauthsession tpm2_verifysignature
+    inst /usr/lib/x86_64-linux-gnu/libtss2-tcti-device.so.0
```
Comment (smoser) [inline layers/build-krd/dracut/soci/module-setup.sh:39]: I suspect this this get dlopen'd ? :-(

File: layers/build-krd/dracut/soci/module-setup.sh [hunk 2]
```diff
@@ -31,6 +31,12 @@ install() {
     inst "/lib/udev/cdrom_id"
     inst "/lib/udev/rules.d/60-cdrom_id.rules"
     inst mknod
+    inst_multiple \
+        tpm2_createprimary tpm2_flushcontext tpm2_getcap tpm2_loadexternal \
```
Comment (smoser) [inline layers/build-krd/dracut/soci/module-setup.sh:35]: it looks to me like 'tpm2' in 20.04 works like lvm does, int hat you don't have to have the '_subcommand' things installed.

So I think you should be able to just do: `inst tpm2`.

File: layers/build-krd/dracut/soci/soci-settled.sh [hunk 3]
```diff
@@ -192,6 +194,29 @@ soci_udev_settled() {
             return 1
         fi
 
+        action=$(<${rootd}/mos-action)
```
Comment (smoser) [inline layers/build-krd/dracut/soci/soci-settled.sh:222]: You're effectively trusting the signer of the manifest to tell you whether or not to extend PCR7 (and thus stop them from having access to the protected TPM values).

it feels to me like that isn't right.

**Review by smoser — COMMENTED** [review]
I guess it is implied that github.com/canonical/go-tpm2 can't do this ?

I'd rather have thing like this done in go than in shell if  possible.

File: layers/build-krd/dracut/soci/module-setup.sh (hunk 1 above)
Comment (hallyn) [reply layers/build-krd/dracut/soci/module-setup.sh:39]: Actually I'm not 100% sure that it's needed.  But it's being used in the atomix installer, so I suspect it is, or we wouldn't have thought to put it there.

File: layers/build-krd/dracut/soci/soci-settled.sh (hunk 3 above)
Comment (hallyn) [reply layers/build-krd/dracut/soci/soci-settled.sh:222]: As I said last night, this is a stop-gap until we get a proper mechanism through the root=soci: replacement.

But we can't block all of the installer work on that.

But as for trusting the the signer of the manifest - so long as we have the luks passwords in keyring, and the SUDI key in /factory/secure tmpfs, it *is* the fact that the signer of manifest can get all of that.

File: layers/build-krd/dracut/soci/module-setup.sh (hunk 2 above)
Comment (hallyn) [reply layers/build-krd/dracut/soci/module-setup.sh:35]: That (inst tpm2) would have been great - tried it, did not work :(

File: layers/build-krd/dracut/soci/soci-settled.sh (hunk 3 above)
Comment (hallyn) [reply layers/build-krd/dracut/soci/soci-settled.sh:222]: I opened an issue to discuss hte better way to do this.

**Comment by hallyn** [comment]
I pulled out the last commit which was trying to make booting from an installed system work.  With this (and the related PRs in trust and mos) provision and install work.  I'll think tonight about how best to get to booting the installed system.

**Comment by hallyn** [comment]
> I guess it is implied that github.com/canonical/go-tpm2 can't do this ?
> 
> I'd rather have thing like this done in go than in shell if possible.

Indeed.  It *should* be possible, and in the background we do still want to switch to that, however something about the EA policy handling is different there.  We are pursuing that route in https://github.com/project-machine/trust/tree/go-tpm2 .  If we can get that working, then we can drop all of these tpm2_* binary dependencies.

**Pull request merged**